src/state/ingress_per_unit.py

```diff
@@ -87,4 +92,9 @@ def from_provider(
                     raise IngressPerUnitIntegrationDataValidationError(
                         "Validation of ingress per unit relation data failed."
                     ) from exc
+                except ValueError as exc:
+                    logger.error(str(exc))
+                    raise IngressPerUnitIntegrationDataValidationError(
+                        "App port from the ingress per unit relation is invalid."
```

Comment:
We'll probably face issues later on with this message not being updated if additional ValueErrors are possible. I'd set a more generic message